plugins/frontend/backstage-plugin-github-insights/src/hooks/useComplianceHooks.ts

```diff
@@ -48,13 +55,20 @@ export const useProtectedBranches = (entity: Entity) => {
   };
 };
 
-export const useRepoLicence = (entity: Entity) => {
+export const useRepoLicence = ({
+  entity,
+  isPrivate = false,
+}: {
+  entity: Entity;
+  isPrivate?: boolean;
+}) => {
   const auth = useApi(githubAuthApiRef);
   const { baseUrl } = useEntityGithubScmIntegration(entity);
   const { owner, repo } = useProjectEntity(entity);
   const { value, loading, error } = useAsync(async (): Promise<any> => {
-    const token = await auth.getAccessToken(['repo']);
-    const octokit = new Octokit({ auth: token });
+    const octokit = new Octokit({
+      auth: isPrivate ? await auth.getAccessToken(['repo']) : undefined,

```

Comment:
Can we check if the user is logged in here? We don't want them to access anonymously if they don't need to (rate limit)

Reply:
Ah, good point 👍🏼